tests/manage/z_cluster/nodes/test_nodes_restart_ms.py

```diff
@@ -0,0 +1,173 @@
+import logging
+import pytest
+import random
+
+
+from ocs_ci.framework.testlib import (
+    tier4a,
+    tier4b,
+    ignore_leftovers,
+    ManageTest,
+    bugzilla,
+    managed_service_required,
+)
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.node import (
+    get_node_objs,
+    recover_node_to_ready_state,
+    get_osd_running_nodes,
+    get_node_osd_ids,
+    wait_for_osd_ids_come_up_on_node,
+    wait_for_nodes_status,
+    verify_worker_nodes_security_groups,
+    get_nodes,
+)
+from ocs_ci.ocs.resources import pod
+from ocs_ci.helpers.sanity_helpers import Sanity
+from ocs_ci.ocs.cluster import (
+    ceph_health_check,
+    is_ms_consumer_cluster,
+)
+from ocs_ci.framework import config
+
+
+logger = logging.getLogger(__name__)
+
+
+@ignore_leftovers
+@managed_service_required
+class TestNodesRestartMS(ManageTest):
+    """
+    Test nodes restart scenarios when using managed service
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup(self, create_scale_pods_and_pvcs_using_kube_job_on_ms_consumers):
+        """
+        Initialize Sanity instance, and create pods and PVCs factory
+
+        """
+        self.sanity_helpers = Sanity()
+        self.create_pods_and_pvcs_factory = (
+            create_scale_pods_and_pvcs_using_kube_job_on_ms_consumers
+        )
+
+    def create_resources(self):
+        """
+        Create resources on the consumers and run IO
+
+        """
+        if is_ms_consumer_cluster():
+            consumer_indexes = [config.cur_index]
+        else:
+            consumer_indexes = config.get_consumer_indexes_list()
```

Comment:
I'm not sure, if I didn't miss anything here, but if I'll have following configuration: 1 provider cluster `pr` and 1 consumer clusters `c1` and `c1` and if I'll run this test with clusters specified in following order: `pr`, `c1`, `c2` (first one is the default/current cluster - provider) - then the `consumer_indexes` will contain index for both consumers (`[1, 2]`), but if I'll run the tests with clusters specified in order `c1`, `c2`, `pr` (default/current cluster will be first consumer), then `consumer_indexes` will contain only list with one item - the default cluster - first consumer. Is this expected behavior?

Reply:
Yes. The test applies to a consumer and provider. So if I run on a consumer, I will create the resources only on the current consumer, and if I run the test on a provider, I will create the resources on all the consumers.

Reply:
I'm not sure, if it is the best approach to change the behavior based on the order of clusters configuration provided during execution. The question is, if it is sufficient to run the tests with one clusters configuration order?

Reply:
Maybe I didn't get you. The test is applicable for both the MS consumer cluster and MS provider cluster. - this means there are two options here(like we can test with vSphere UPI and vSphere LSO).

So suppose we want to test the MS consumer cluster, we will create the resources only on the current consumer cluster(because the other consumers are not relevant).
And if we run the test on a provider, we will create the resources on all the consumer clusters.

Let me know if it makes sense.

Reply:
Thanks for the explanation.